cmd/litewitness/litewitness.go

```diff
@@ -279,7 +380,16 @@ func connectToBastion(ctx context.Context, bastion string, signer *signer, srv *
 		slog.Info("connecting to bastion failed", "bastion", bastion, "err", err)
 		return fmt.Errorf("connecting to bastion: %v", err)
 	}
+	// Ensure that the connection is closed when our context is cancelled.
+	go func() {
```

Comment:
The function can return even if the context is not cancelled, leaking the goroutine.